agent/src/containerutilityinterface.cpp

```diff
@@ -70,4 +78,21 @@ void ContainerUtilityInterface::removeOldContainers()
     delete containerNames;
 }
 
+void ContainerUtilityInterface::checkWorkspace()
+{
+    const std::string rootDir = m_config->getStringValue("SoftwareContainer", "shared-mounts-dir");
+    if (!isDirectory(rootDir)) {
+        log_debug() << "Container root " << rootDir << " does not exist, trying to create";
+        std::unique_ptr<CreateDir> createDirInstance{new CreateDir()};
```

Comment:
Copy paste from Kursats comment before, it's just a syntax difference in this case I think. It does look nicer than the original way of writing it however.